src/webargs/falconparser.py

```diff
@@ -35,17 +37,20 @@ def parse_form_body(req: falcon.Request):
         req.content_type is not None
         and "application/x-www-form-urlencoded" in req.content_type
     ):
-        body = req.stream.read(req.content_length or 0)
-        try:
-            body = body.decode("ascii")
-        except UnicodeDecodeError:
-            body = None
-            req.log_error(
-                "Non-ASCII characters found in form body "
-                "with Content-Type of "
-                "application/x-www-form-urlencoded. Body "
-                "will be ignored."
-            )
+        # the type of `req.stream.read()` is `str | Any` according to annotations
+        # but at runtime it's always `bytes`, so coerce by checking the type
+        body: str | bytes | None = req.stream.read(req.content_length or 0)
+        if isinstance(body, bytes):
```

Comment:
I don't understand why this is better than e.g. a cast or and ignore.

Reply:
Let me double-check this before we consider merging. At the time that I was touching this, I was under the impression that this `read()` could sometimes return a `str`, but now I'm no longer sure.

Reply:
Let me turn this into a type-ignore.

I have tried a few specific things against the latest `falcon` version, and I can't make `req.stream.read` return anything other than bytes.

Thanks for calling this out!

Reply:
Hi there :wave:

Could you elaborate on when reading [`req.stream`](https://falcon.readthedocs.io/en/stable/api/request_and_response_wsgi.html#falcon.Request.stream) yields `str` or `None`?

This is not something we statically declare either:
```python
import typing

import falcon
import falcon.testing


def main() -> None:
    req = falcon.testing.create_req()
    typing.reveal_type(req.stream.read)


if __name__ == '__main__':
    main()
```
:arrow_down: 
```
$ mypy --strict test.py
test.py:9: note: Revealed type is "def (builtins.int | None =) -> builtins.bytes"
Success: no issues found in 1 source file
```

Reply:
Hi, thanks for reaching out to a downstream project!

I recall with some clarity that I looked at the `falcon` source at the time and couldn't find an obvious explanation, but that `mypy` was flagging this type as `bytes | str | Any` when I was running it. It didn't make a ton of sense to me, but I couldn't quickly rule out the possibility that I was not understanding the `falcon` source.

I just tried checking things out with stuff pinned back to `falcon<4.1`, `mypy==1.11.0`, and even `typing_extensions==4.0`, and I can't reproduce it. So I don't think there's an upstream issue at all!

We could update this block of code to drop the try-decode step, but I'd want to check that against very old `falcon` versions too.

Reply:
You're welcome!

Hm, reading from the input stream has always been `bytes`, I don't think you need to check anything.

One possible reason could be that _webargs_ still support Falcon 2.0, which in turn still supported Python 2.7 back in the day where `str` and `bytes` were synonymous (instead you could explicitly use `unicode` strings). Maybe that's where the confusion is stemming from? 
However, if your project no longer supports Python 2.7, you can safely ignore this possibility altogether.

Otherwise you can safely switch to just `bytes` in #725.